### PR TITLE
Wallaby.Selenium.upload_file/2 needs to allow cookies 🍪

### PR DESCRIPTION
**Pull request by hazen**
- Clean up Dialyzer errors

https://app.clubhouse.io/payitoff/story/3348/file-uploads-in-wallaby-via-selenium-be-borked

monster

### CODE REVIEW

File: lib/wallaby/browser.ex [hunk 1]
```diff
@@ -542,7 +542,6 @@ defmodule Wallaby.Browser do
     end
   end
 
-  @spec click(parent, Query.t()) :: parent
```
Comment (hazen) [inline lib/wallaby/browser.ex:545]: Duplicated `@spec`

File: lib/wallaby/httpclient.ex [hunk 2]
```diff
@@ -21,7 +21,7 @@ defmodule Wallaby.HTTPClient do
   response.
   """
   @spec request(method, url, params, request_opts) ::
-          {:ok, response}
+          {:ok, response, cookies()}
```
Comment (hazen) [inline lib/wallaby/httpclient.ex:24]: Yes, this is now always returned

File: lib/wallaby/selenium.ex [hunk 3]
```diff
@@ -386,8 +386,12 @@ defmodule Wallaby.Selenium do
     zip64 = make_file(filename)
     endpoint = element.session_url <> "/file"
 
-    with {:ok, response} <- Wallaby.HTTPClient.request(:post, endpoint, %{file: zip64}) do
-      Map.fetch!(response, "value")
+    case Wallaby.HTTPClient.request(:post, endpoint, %{file: zip64}) do
+      {:ok, response, _cookies} ->
```
Comment (hazen) [inline lib/wallaby/selenium.ex:390]: 🥩 The fix! Match on the third argument

File: lib/wallaby/webdriver_client.ex [hunk 4]
```diff
@@ -658,7 +658,8 @@ defmodule Wallaby.WebdriverClient do
   end
 
   # Pull the cookies from the element if the session is nil
-  @spec resolve_cookies(Session.t() | nil, Element.t()) :: cookies()
+  @spec resolve_cookies(Session.t() | nil, Element.t() | nil) :: cookies()
```
Comment (hazen) [inline lib/wallaby/webdriver_client.ex:661]: Handle case in which session and element are both `nil`

**Review by JEG2 — APPROVED** [review]
Still thrilled this (so barely) wasn't me!  😆

**Review by bobwaycott — CHANGES_REQUESTED** [review]
seems fine to me ... but i'm still wondering why the other uploads have worked? do they still work?

**Change request:** bump the version in mix file for a new release

**Review by bobwaycott — APPROVED** [review]
word

**Pull request merged**